actors/miner/tests/extend_sector_expiration_test.rs

```diff
@@ -0,0 +1,367 @@
+    let params = ExtendSectorExpirationParams { extensions };
+
+    h.extend_sectors(&mut rt, params).unwrap();
+    let state: State = rt.get_state();
+    let deadlines = state.load_deadlines(rt.store()).unwrap();
+
+    // half of the sectors should expire on-time
+    let mut on_time_total = 0;
+    deadlines
+        .for_each(&policy, rt.store(), |deadline_index, mut deadline| {
+            let expiration_set = deadline
+                .pop_expired_sectors(
+                    rt.store(),
+                    new_expiration - 1,
+                    state.quant_spec_for_deadline(&policy, deadline_index),
+                )
+                .unwrap();
+            on_time_total += expiration_set.len();
+            Ok(())
+        })
+        .unwrap();
+    assert_eq!(sector_count / 2, on_time_total);
+
+    // half of the sectors should expire late
+    let mut extended_total = 0;
+    deadlines
+        .for_each(&policy, rt.store(), |deadline_index, mut deadline| {
+            let expiration_set = deadline
+                .pop_expired_sectors(
+                    rt.store(),
+                    new_expiration - 1,
+                    state.quant_spec_for_deadline(&policy, deadline_index),
+                )
+                .unwrap();
+            extended_total += expiration_set.len();
+            Ok(())
+        })
+        .unwrap();
+    assert_eq!(sector_count / 2, extended_total);
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn supports_extensions_off_deadline_boundary() {
+    let (mut h, mut rt) = setup();
+    let old_sector = commit_sector(&mut h, &mut rt);
+    h.advance_and_submit_posts(&mut rt, &vec![old_sector.clone()]);
+
+    let state: State = rt.get_state();
+    let policy = Policy::default();
+    let (deadline_index, partition_index) =
+        state.find_sector(&policy, rt.store(), old_sector.sector_number).unwrap();
+
+    let extension = 42 * policy.wpost_proving_period + policy.wpost_proving_period / 3;
+    let new_expiration = old_sector.expiration + extension;
+
+    let params = ExtendSectorExpirationParams {
+        extensions: vec![ExpirationExtension {
+            deadline: deadline_index,
+            partition: partition_index,
+            sectors: make_bitfield(&[old_sector.sector_number]),
+            new_expiration,
+        }],
+    };
+
+    h.extend_sectors(&mut rt, params).unwrap();
+
+    // assert sector expiration is set to the new value
+    let mut state: State = rt.get_state();
+    let new_sector = h.get_sector(&rt, old_sector.sector_number);
+    assert_eq!(new_expiration, new_sector.expiration);
+
+    // advance clock to expiration
+    rt.set_epoch(new_sector.expiration);
+    state.proving_period_start += policy.wpost_proving_period
+        * ((rt.epoch - state.proving_period_start) / policy.wpost_proving_period + 1);
+    rt.replace_state(&state);
+
+    // confirm it is not in sector's deadline
```

Comment:
nit: for clarity "to make sure we're testing extensions off the deadline boundary"